src/samples/OrientedBoundingBox.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Rock-Core
+#ifndef BASE_SAMPLES_ORIENTEDBOUNDINGBOX_HPP_
+#define BASE_SAMPLES_ORIENTEDBOUNDINGBOX_HPP_
+
+#include <base/Pose.hpp>
+#include <base/samples/BoundingBox.hpp>
+
+namespace base {
+namespace samples {
+/** Oriented Bounding Box (OBB) representation.
+ *
+ * The data structure represents a three-dimensional Oriented Bounding Box
+ * (OBB) with the associated uncertainty. It is derived from
+ * base::samples::BoundingBox and includes the rotation information with
+ * respect to the bounding box center position.
+ *
+ * @see base::samples::BoundingBox
+ */
+struct OrientedBoundingBox : public BoundingBox {
+    OrientedBoundingBox();
+
+    OrientedBoundingBox(const base::Time& time,
+                        const base::Vector3d& position);
+
+    OrientedBoundingBox(const base::Time& time,
+                        const base::Vector3d& position,
+                        const base::Vector3d& dimension);
+
+    OrientedBoundingBox(const base::Time& time,
+                        const base::Vector3d& position,
+                        const base::Vector3d& dimension,
+                        const base::Vector4d& orientation);
+
+    OrientedBoundingBox(const base::Time& time,
+                        const base::Vector3d& position,
+                        const base::Vector3d& dimension,
+                        const base::Orientation& orientation);
```

Comment:
Any reason why you don't use default arguments instead of overloads, which would allow to have a single constructor ... (Just remember to make it explicit).

Reply:
No reason that I can remember now. I am going to change for default arguments.

Reply:
I've changed as you suggested. I replaced the multiple overloaded constructors by a single one with default arguments.